permissions/src/main/java/com/google/accompanist/permissions/MutablePermissionState.kt

```diff
@@ -86,26 +86,41 @@ internal class MutablePermissionState(
     private val activity: Activity
 ) : PermissionState {
 
-    override var status: PermissionStatus by mutableStateOf(getPermissionStatus())
+    override var status: PermissionStatus by mutableStateOf(getPermissionStatus(false))
 
     override fun launchPermissionRequest() {
         launcher?.launch(
             permission
         ) ?: throw IllegalStateException("ActivityResultLauncher cannot be null")
     }
 
+    override fun openAppSettings() {
+        activity.openAppSettings(permission)
+    }
+
     internal var launcher: ActivityResultLauncher<String>? = null
 
     internal fun refreshPermissionStatus() {
-        status = getPermissionStatus()
+        status = getPermissionStatus(isPermissionPostRequest = true)
     }
 
-    private fun getPermissionStatus(): PermissionStatus {
+    private fun getPermissionStatus(isPermissionPostRequest: Boolean): PermissionStatus {
         val hasPermission = context.checkPermission(permission)
         return if (hasPermission) {
             PermissionStatus.Granted
         } else {
-            PermissionStatus.Denied(activity.shouldShowRationale(permission))
+            val shouldShowRationale = activity.shouldShowRationale(permission)
+            when {
+                isPermissionPostRequest -> when {
+                    shouldShowRationale -> PermissionStatus.NotGranted.Denied
+                    else -> PermissionStatus.NotGranted.PermanentlyDenied
```

Comment:
There's a case here where it's possible to have requested the permission and `shouldShowRationale` is false, but without the permission being permanently denied.

If a permission is requested for the first time, and the user uses the back button to exit the permission dialog without selecting any of the options, the permission won't be permanently denied. If the permission is requested again, then the dialog will be shown again normally. Also, since the user did not explicitly deny the permission, `shouldShowRationale` will still be false.